Lib/pyclbr.py

```diff
@@ -174,17 +198,22 @@ def _readmodule(module, path, inpackage=None):
                 tokentype, meth_name, start = next(g)[0:3]
                 if tokentype != NAME:
                     continue # Syntax error
+                cur_func = None
                 if stack:
-                    cur_class = stack[-1][0]
-                    if isinstance(cur_class, Class):
-                        # it's a method
-                        cur_class._addmethod(meth_name, lineno)
-                    # else it's a nested def
+                    cur_obj = stack[-1][0]
+                    if isinstance(cur_obj, Object):
```

Comment:
I think stack can reach None:
```
while stack and stack[-1][1] >= thisindent:
    del stack[-1]
```

I'm going to commit the other changes for review and then look at this more.  There's also a lot of duplicated code between the `elif token == 'def':` and `elif token == 'class':` branches.  The main difference is inheritance for the `class`.  I didn't know how much you wanted to change existing code.